packages/db/src/core/cli/migration-queries.ts

```diff
@@ -40,12 +40,23 @@ const genTempTableName = customAlphabet('abcdefghijklmnopqrstuvwxyz', 10);
 export async function getMigrationQueries({
 	oldSnapshot,
 	newSnapshot,
+	reset = false
 }: {
 	oldSnapshot: DBSnapshot;
 	newSnapshot: DBSnapshot;
+	reset?: boolean;
 }): Promise<{ queries: string[]; confirmations: string[] }> {
-	const queries: string[] = [];
+	let queries: string[] = [];
 	const confirmations: string[] = [];
+
+	// When doing a reset, first create DROP TABLE statements, then treat everything
+	// else as creation.
+	if(reset) {
+		const curSnapshot = oldSnapshot;
```

Comment:
nit `current` > `cur`

Reply:
actually, this name is a bit confusing, since it's not really the current snapshot right? Maybe like `originalOldSnapshot`?